### ISSUE TITLE
[markdown-mode] Random syntax highlighting / font styles

### ISSUE THREAD

**Issue by Martinsos** [syl20bnr/spacemacs#16170]
When in markdown mode, font styles and syntax coloring get applied for some markdown elements, for some not, and for some are broken.

::::
TLDR; I explain how I tested at the bottom, but this seems to be caused directly by the `markdown` spacemacs layer (because it works if I use markdown-mode as additional package).
::::

This happens to me for even very simple examples.

By doing small modifications, like adding a whitespace and then removing it, I get coloring/styles to refresh and I end up with some other random coloring/styles -> so each time I get something else.

For example, here I managed to get three different colorings for this file (none of them completely correct) by just adding and removing lines to the file (always ending up at the same state). So I would add and remove the empty line, and I get different coloring:

![image](https://github.com/jrblevin/markdown-mode/assets/1536647/99785b31-054a-45a1-98ee-e972e826974c)

![image](https://github.com/jrblevin/markdown-mode/assets/1536647/40f580ad-e800-4ade-8a2e-4b2477b8c64e)

![image](https://github.com/jrblevin/markdown-mode/assets/1536647/19b99236-5d10-450d-b266-d140d72e5307)

This doesn't happen just for backticks, it also happens for example for bold text : `**text**` -> sometimes it gets bolded, sometimes it doesn't. For URLs also `[link](https://test.com)`.

On following photo bold works but URL and backticks don't.

![image](https://github.com/jrblevin/markdown-mode/assets/1536647/5b39f1f2-b5fa-4eb1-bb3d-b5b0920faff9)

This is happening on Archlinux, emacs 29.1, newest spacemacs from `develop` branch, markdown-mode 2.7-alpha (I updated all the packages).

```
Minor modes enabled in this buffer: Anzu Auto-Save Company Eldoc Emoji-Cheat-Sheet-Plus-Display
Evil-Local Evil-Mc Evil-Surround Font-Lock Hl-Todo Spacemacs-Leader-Override Subword Undo-Tree
Vi-Tilde-Fringe Yas
```

I tried turning off all modes except for `Font-Lock` and `Spacemacs-Leader-Override` and it didn't help, problem is still here.

I also tried running emacs with `emacs -Q --load ~/local/path/to/markdown-mode.el test-case.md` and that works fine, results in markdown-mode that works ok.

I also tried deleting my whole .spacemacs.d/ and starting with spacemacs from fresh, so fresh config, where I picked evil mode and added only markdown layer, and again I get the issue described here -> so it doesn't seem to be my configuration causing it.

Finally, I tried commenting out `markdown` layer and adding `markdown-mode` under `additional-packages` -> and that worked! So if I use `markdown-mode` directly, all is ok. But if I use `markdown` layer, then I have these issues. So my current conclusion is that this is somehow caused by `markdown` layer.

**Comment by real-or-random** [comment]
I can confirm this issue with markdown-mode. It's rather annoying. 

> So my current conclusion is that this is somehow caused by `markdown` layer.

Indeed, and the issue disappears when I disable `mmm-mode`!

**Comment by real-or-random** [comment]
I can't try this in spacemacs right now. (Sorry, I've switched to doom... ), but now that the mode is removed, a partial replacement should be to set `markdown-fontify-code-blocks-natively`.  This will at least enable syntax highlighting in an embedded code block, and it should be much less issue-prone than mmm-mode.

**Comment by smile13241324** [comment]
This is done already in markdowns init function, I think that's the reason why I haven't noticed any changes in my environment.
